Kanan/Currtarget.cpp

```diff
@@ -0,0 +1,256 @@
+#pragma once
+#define IMGUI_DEFINE_MATH_OPERATORS
+#include <imgui.h>
+#include <imgui_internal.h>
+
+#include "Mabinogi.hpp"
+#include "KCharacter.hpp"
+#include "Kanan.hpp"
+#include "Currtarget.hpp"
+
+namespace kanan {
+
+	//most of this is based off status ui and entity viewer
+
+
+
+	Currtarget::Currtarget()
+		: m_hp{}
+	{
+		m_hp.reserve(100);
+	}
+
+	void Currtarget::onFrame()
+	{
+
+		auto game = g_kanan->getGame();
+		auto localCharacter = game->getLocalCharacter();
+		KCharacter* target = nullptr;
+
+		//check if there is a local char or target for local char. if not dont bother rendering window
+		if (!localCharacter) { hastarget = false; }
+		else { hastarget = true; }
+		if (localCharacter) {
+			target = g_kanan->getGame()->getCharacterByID(localCharacter->targetID);
+			if (target == nullptr) { hastarget = false; }
+			else { hastarget = true; }
+		}
+
+
+		if (g_kanan->isUIOpen()) { hastarget = true; }
+
+		//if enabled create new window for details
+		if (m_is_enabled && hastarget) {
+			createtargetwindow();
+		}
+	}
+
+	void Currtarget::onUI()
+	{
+		if (ImGui::CollapsingHeader("Show details of targeted mob")) {
+			if (ImGui::Checkbox("Show details", &m_is_enabled)) {}
+			if (m_is_enabled) {
+				if (ImGui::Checkbox("Show Target Pos", &showpos)) {}
+				if (ImGui::Checkbox("Show Targets, Target", &showothertarget)) {}
+				if (ImGui::Checkbox("Show Target HP", &showtargethp)) {}
+			}
+		}
+
+
+
+
+	}
+
+
+
+	//create imgui window for details of target to exist in
+	void Currtarget::createtargetwindow() {
+		// Clear the current list.
+		m_characters.clear();
+		auto game = g_kanan->getGame();
+		//get local char
+		auto localCharacter = game->getLocalCharacter();
+
+
+		//if no local char do nothing
+
+		//get target of local char
+		KCharacter* target = nullptr;
+		KCharacter* targetchar = nullptr;
+
+		//if we have a localCharacter do rest of logic, if not do nothing
+		if (localCharacter) {
+			// Iterate over every character and add it to our list.
+			target = g_kanan->getGame()->getCharacterByID(localCharacter->targetID);
+			auto entityList = game->getEntityList();
+			//get all other characters
+			auto& characters = entityList->characters;
+			auto highestIndex = characters.count;
+			auto node = characters.root;
+
+
+
+			//create a list of all local chars
+
+			for (uint32_t i = 0; i <= highestIndex && node != nullptr; ++i, node = node->next) {
```

Comment:
This loop seems unnecessary, you've already gotten the target (`target`) and don't use `m_characters` for anything else.